### DISCUSSION TITLE
PostgreSQL version upgrade

### DISCUSSION THREAD

**Discussion by rubenskuhl** [irrdnet/irrd#981] [Q&A]
Any concerns or pitfalls upgrading PostgreSQL 12 to a newer one in an IRRd 4.4.4 environment ? 
For now, the only thing I mapped is that P12 used MD5 and will require password reset in a newer version. 
IRRd mentions "Versions 9.6, 11.16, 13.7, 15.0" - Should I not got beyond 15 ? 16 is the one used in newer distributions, while 17 is available. Isn't 14 tested/recommended ?

**Comment by mxsasha** [comment]
There are no known issues with newer versions of PostgreSQL. Perhaps some of the configuration advice could be revisited, like whether setting `random_page_cost` different from default still applies. The specific list comes from the [version matrix in CI configuration](https://github.com/irrdnet/irrd/blob/main/.circleci/config.yml#L52-L54) which is run before release. For brevity, I don't test every intermediate version of all dependencies there.
